app/controllers/v2/application_controller.rb

```diff
@@ -60,5 +60,17 @@ def rbac_allowed?
       permission = self.class.instance_variable_get(:@action_permissions)[action_name.to_sym]
       user.authorized_to?(Rbac::INVENTORY_HOSTS_READ) && user.authorized_to?(permission)
     end
+
+    # Reduce through all the parents of the resource and join+scope them on the resource
+    def expand_resource
+      permitted_params[:parents].to_a.reduce(resource) do |scope, parent|
+        ref = scope.reflect_on_association(parent)
+        klass = ref.klass
+
+        scope.joins(parent)
+             .where(parent => { klass.primary_key => permitted_params[ref.foreign_key] })
```

Comment:
Similar here, can `permitted_params[ref.foreign_key]` be `nil`?

Reply:
This won't be called if the ID is `nil` also this behavior is given by rails.